libraries/exoplayer_hls/src/main/java/androidx/media3/exoplayer/hls/playlist/DefaultHlsPlaylistTracker.java

```diff
@@ -141,6 +146,25 @@ public void start(
             playlistParserFactory.createPlaylistParser());
     Assertions.checkState(initialPlaylistLoader == null);
     initialPlaylistLoader = new Loader("DefaultHlsPlaylistTracker:MultivariantPlaylist");
+    SntpClient.InitializationCallback callback =
+        new SntpClient.InitializationCallback() {
+          @Override
+          public void onInitialized() {
+            requestMasterPlaylist(eventDispatcher, multivariantPlaylistLoadable);
+          }
+
+          @SuppressLint("Range")
+          @Override
+          public void onInitializationFailed(IOException error) {
+            Log.w(TAG, "NTP time init failed, use default system time", error);
+            requestMasterPlaylist(eventDispatcher, multivariantPlaylistLoadable);
+          }
+        };
+    SntpClient.initialize(initialPlaylistLoader, callback);
+  }
+
+  private void requestMasterPlaylist(
+      EventDispatcher eventDispatcher, ParsingLoadable<HlsPlaylist> multivariantPlaylistLoadable) {
     long elapsedRealtime =
         initialPlaylistLoader.startLoading(
             multivariantPlaylistLoadable,
```

Comment:
The `requestMasterPlaylist` was simply a refactoring (extract method), then call the method in the `SntpClient` callback.  Most times this is a no-op as the `SntpClient` offset is up to date.